pybedlite/overlap_detector.py

```diff
@@ -322,18 +322,38 @@ def overlaps_any(self, interval: Span) -> bool:
             True if and only if the given interval overlaps with any interval in this
             detector.
         """
-        tree = self._refname_to_tree.get(interval.refname)
+        tree = self._refname_to_tree.get(interval.refname, None)
         if tree is None:
             return False
         else:
             if not self._refname_to_indexed[interval.refname]:
                 tree.index()
-            try:
-                next(iter(tree.overlap(interval.refname, interval.start, interval.end)))
-            except StopIteration:
-                return False
-            else:
-                return True
+                self._refname_to_indexed[interval.refname] = True
+            # IntervalSet uses closed intervals whereas we are using half-open intervals, so add 1
+            # to start
+            return tree.any_overlaps(interval.start + 1, interval.end)
+
+    def iter_overlaps(self, interval: Span) -> Iterator[SpanType]:
+        """Yields any intervals in this detector that overlap the given interval
+
+        Args:
+            interval: the interval to check
+
+        Yields:
+            Intervals in this detector that overlap the given interval, in insertion order.
+        """
+        tree = self._refname_to_tree.get(interval.refname, None)
+        if tree is not None:
+            if not self._refname_to_indexed[interval.refname]:
+                tree.index()
+                self._refname_to_indexed[interval.refname] = True
+            ref_intervals: List[SpanType] = self._refname_to_intervals[interval.refname]
+            # IntervalSet uses closed intervals whereas we are using half-open intervals, so add 1
+            # to start.
+            # Also IntervalSet yields indices in reverse insertion order, so yield intervals in
+            # reverse of indices list.
+            for index in reversed(tree.find_overlaps(interval.start + 1, interval.end)):
+                yield ref_intervals[index]
```

Comment:
The call to `reversed` won't make this lazy despite its use in a generator.

Are you doing this to preserve original order?

IMHO order wouldn't matter to me.

Or if it did, I'd want the same order as is forced in `get_overlaps()` further down.

Reply:
I put the call to `reversed` to yield them in insertion order. The `IntervalSet` returns a list in reverse-insertion order, and `reversed` iterates backwards through the list without copying, so basically no overhead. I don't feel strongly, the point of this method was to have a lightweight method for yielding all the intervals without concern for order.